deepchem/data/datasets.py

```diff
@@ -464,12 +462,27 @@ def create_dataset(shard_generator, data_dir=None, tasks=[], verbose=True):
           DiskDataset.write_data_to_disk(data_dir, basename, tasks, X, y, w,
                                          ids))
     metadata_df = DiskDataset._construct_metadata(metadata_rows)
-    metadata_filename = os.path.join(data_dir, "metadata.joblib")
-    save_to_disk((tasks, metadata_df), metadata_filename)
+    save_metadata(tasks, metadata_df, data_dir)
     time2 = time.time()
     log("TIMING: dataset construction took %0.3f s" % (time2 - time1), verbose)
     return DiskDataset(data_dir, verbose=verbose)
 
+  def load_metadata(self):
+    try:
+      tasks_filename, metadata_filename = self._get_metadata_filename()
+      with open(tasks_filename) as fin:
+        tasks = json.load(fin)
+      metadata_df = read_hdf(metadata_filename, 'metadata')
+      return tasks, metadata_df
+    except Exception as e:
+      pass
+
+    metadata_filename = os.path.join(self.data_dir, "metadata.joblib")
```

Comment:
Maybe add a comment at this point to explain what it's doing?  "The metadata is stored in an obsolete format, so upgrade it to the newer format."

Reply:
+1